js_modules/dagster-ui/packages/ui-core/src/overview/useGroupTimelineRunsBy.tsx

```diff
@@ -23,13 +23,33 @@ export const useGroupTimelineRunsBy = (
     [defaultValue],
   );
 
-  const [groupRunsBy, setGroupRunsBy] = useStateWithStorage(storageKey, validate);
+  const history = useHistory();
+  const [groupRunsBy, setGroupRunsBy] = useQueryAndLocalStoragePersistedState<GroupRunsBy>({
+    localStorageKey: GROUP_BY_KEY,
+    queryKey: 'groupBy',
+    encode: (value) => {
+      return {groupBy: value};
+    },
+    decode: (pair) => {
+      return validate(pair.groupBy);
+    },
+    isEmptyState: (value) => !value,
+  });
+
   const setGroupByWithDefault = useCallback(
     (value: GroupRunsBy) => {
       setGroupRunsBy(value || defaultValue);
     },
-    [defaultValue, setGroupRunsBy],
+    [defaultValue, setGroupRunsBy, history],
   );
 
+  // Runs on initial render to ensure groupBy query param is set.
+  useEffect(() => {
+    const searchParams = new URLSearchParams(location.search);
+    if (!searchParams.has('groupBy')) {
+      setGroupByWithDefault(defaultValue);
+    }
+  }, [location.search, setGroupByWithDefault, defaultValue]);
```

Comment:
I don't think you need this. If it's not set you can just return the default value from the `decode` function